src/main/telemetry/ibus.c

```diff
@@ -0,0 +1,413 @@
+};
+
+static serialPort_t *ibusSerialPort = NULL;
+static serialPortConfig_t *ibusSerialPortConfig;
+
+static bool ibusTelemetryEnabled = false;
+static portSharing_e ibusPortSharing;
+
+#define INVALID_IBUS_ADDRESS 0
+static ibusAddress_t ibusBaseAddress = INVALID_IBUS_ADDRESS;
+
+static uint8_t ibusReceiveBuffer[IBUS_RX_BUF_LEN] = { 0x0 };
+
+static uint16_t calculateChecksum(const uint8_t *ibusPacket, size_t packetLength)
+{
+    uint16_t checksum = 0xFFFF;
+    for (size_t i = 0; i < packetLength - IBUS_CHECKSUM_SIZE; i++) {
+        checksum -= ibusPacket[i];
+    }
+
+    return checksum;
+}
+
+static bool isChecksumOk(const uint8_t *ibusPacket)
+{
+    uint16_t calculatedChecksum = calculateChecksum(ibusReceiveBuffer, IBUS_RX_BUF_LEN);
+
+    // Note that there's a byte order swap to little endian here
+    return (calculatedChecksum >> 8) == ibusPacket[IBUS_RX_BUF_LEN - 1]
+           && (calculatedChecksum & 0xFF) == ibusPacket[IBUS_RX_BUF_LEN - 2];
+}
+
+static void transmitIbusPacket(uint8_t *ibusPacket, size_t payloadLength)
+{
+    uint16_t checksum = calculateChecksum(ibusPacket, payloadLength + IBUS_CHECKSUM_SIZE);
+    for (size_t i = 0; i < payloadLength; i++) {
+        serialWrite(ibusSerialPort, ibusPacket[i]);
+    }
+    serialWrite(ibusSerialPort, checksum & 0xFF);
+    serialWrite(ibusSerialPort, checksum >> 8);
+}
+
+static void sendIbusDiscoverSensorReply(ibusAddress_t address)
+{
+    uint8_t sendBuffer[] = { 0x04, IBUS_COMMAND_DISCOVER_SENSOR | address};
+    transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+}
+
+static void sendIbusSensorType(ibusAddress_t address)
+{
+    uint8_t sendBuffer[] = {0x06,
+                            IBUS_COMMAND_SENSOR_TYPE | address,
+                            sensorAddressTypeLookup[address - ibusBaseAddress],
+                            0x02
+                           };
+    transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+}
+
+static void sendIbusMeasurement(ibusAddress_t address, uint16_t measurement)
+{
+    uint8_t sendBuffer[] = { 0x06, IBUS_COMMAND_MEASUREMENT | address, measurement & 0xFF, measurement >> 8};
+    transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+}
+
+static bool isCommand(ibusCommand_e expected, const uint8_t *ibusPacket)
+{
+    return (ibusPacket[1] & 0xF0) == expected;
+}
+
+static ibusAddress_t getAddress(const uint8_t *ibusPacket)
+{
+    return (ibusPacket[1] & 0x0F);
+}
+
+static void dispatchMeasurementReply(ibusAddress_t address)
+{
+    int value;
+    
+    switch (sensorAddressTypeLookup[address - ibusBaseAddress]) {
+    case IBUS_SENSOR_TYPE_EXTERNAL_VOLTAGE:
+        value = vbat * 10;
+        if (ibusTelemetryConfig()->report_cell_voltage) {
+            value /= batteryCellCount;
+        }
+        sendIbusMeasurement(address, value);
+        break;
+
+    case IBUS_SENSOR_TYPE_TEMPERATURE:
+        #ifdef BARO
+            value = (baroTemperature + 50) / 10;
```

Comment:
The 0.5C offset is because original code did compute value in degrees. Integer division in C is rounding positive numbers down (190/100 == 1), with this offset resulting (positive) value is correct. Please read my earlier comments ...